app/routes.py

```diff
@@ -22,9 +23,22 @@ def index():
 def create():
     form = LoginForm()
     create_form = ChallengeCreationForm()
+    errors = []
     if create_form.validate_on_submit():
+        if current_user.is_anonymous:
+            errors.append("Login before creating a challenge!")
```

Comment:
Maybe make a rerender immediately after here ? ideally we make this check on switching to create (login required - maybe have to use a JS script to display the modal? out of scope for this issue